From 648b2bf73461c5754e5eaad70897f49e9a492a14 Mon Sep 17 00:00:00 2001
From: Cedric Neveux <cedric.neveux@nxp.com>
Date: Tue, 17 Apr 2018 18:04:39 +0200
Subject: [PATCH 4194/5242] MLK-17909 RNG Instantation done in Secure Firmware

commit  8927e841508b8052eda233e806e7e459a4569035 from
https://source.codeaurora.org/external/imx/linux-imx.git

   - For i.MX 6 and 7 check if the Secure Firmware (OPTEE) is present.
     If present don't do the RNG instantation in the CAAM driver

Reviewed-by: Silvano Di Ninno <silvano.dininno@nxp.com>
Signed-off-by: Cedric Neveux <cedric.neveux@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/inst_rng.c |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/inst_rng.c b/drivers/crypto/caam/inst_rng.c
index a2118e4..ae1846d 100644
--- a/drivers/crypto/caam/inst_rng.c
+++ b/drivers/crypto/caam/inst_rng.c
@@ -2,7 +2,7 @@
 /*
  * CAAM RNG instantiation driver backend
  *
- * Copyright 2017-2018 NXP Semiconductor, Inc.
+ * Copyright 2017-2018 NXP
  */
 
 #include <linux/device.h>
@@ -287,6 +287,17 @@ int inst_rng_imx6(struct platform_device *pdev)
 	ctrlpriv = dev_get_drvdata(ctrldev);
 	ctrl = (struct caam_ctrl __iomem *)ctrlpriv->ctrl;
 
+#ifndef CONFIG_ARM64
+	/*
+	 * Check if the Secure Firmware is running,
+	 * check only for i.MX6 and i.MX7
+	 */
+	if (of_find_compatible_node(NULL, NULL, "linaro,optee-tz")) {
+		pr_info("RNG Instantation done by Secure Firmware\n");
+		return ret;
+	}
+#endif
+
 	cha_vid_ls = rd_reg32(&ctrl->perfmon.cha_id_ls);
 
 	/*
-- 
1.7.9.5

